### DISCUSSION TITLE
wg dashboard start fails

### DISCUSSION THREAD

**Discussion by goerdi** [WGDashboard/WGDashboard#539] [General]
Hi !

i went throug instalation process in Ubuntu 24.04  but start fails
[WGDashboard] Starting WGDashboard in the foreground.
Traceback (most recent call last):
  File "/usr/local/bin/wgdashboard/src/dashboard.py", line 18, in <module>
    import bcrypt
ModuleNotFoundError: No module named 'bcrypt'

But the module is installes...

Ciao Gerd

**Comment by donaldzou** [comment]
@goerdi Hi! Could you please try the following command, assuming you're in `wgdashboard/src`

```
./venv/bin/python3 -m pip install -r requirements.txt
```

**Comment by goerdi** [comment]
Hi !

root@virgo:/usr/local/bin/wgdashboard/src# ./venv/bin/python3 -m pip install -r requirements.txt
Collecting bcrypt (from -r requirements.txt (line 1))
  Using cached bcrypt-4.2.1-cp39-abi3-manylinux_2_28_x86_64.whl.metadata (9.8 kB)
Requirement already satisfied: ifcfg in ./venv/lib/python3.12/site-packages (from -r requirements.txt (line 2)) (0.24)
Collecting psutil (from -r requirements.txt (line 3))
  Using cached psutil-6.1.0-cp36-abi3-manylinux_2_12_x86_64.manylinux2010_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (22 kB)
Collecting pyotp (from -r requirements.txt (line 4))
  Using cached pyotp-2.9.0-py3-none-any.whl.metadata (9.8 kB)
Collecting Flask (from -r requirements.txt (line 5))
  Using cached flask-3.1.0-py3-none-any.whl.metadata (2.7 kB)
Collecting flask-cors (from -r requirements.txt (line 6))
  Using cached Flask_Cors-5.0.0-py2.py3-none-any.whl.metadata (5.5 kB)
Collecting icmplib (from -r requirements.txt (line 7))
  Using cached icmplib-3.0.4-py3-none-any.whl.metadata (19 kB)
Collecting gunicorn (from -r requirements.txt (line 8))
  Using cached gunicorn-23.0.0-py3-none-any.whl.metadata (4.4 kB)
Collecting requests (from -r requirements.txt (line 9))
  Using cached requests-2.32.3-py3-none-any.whl.metadata (4.6 kB)
Collecting Werkzeug>=3.1 (from Flask->-r requirements.txt (line 5))
  Using cached werkzeug-3.1.3-py3-none-any.whl.metadata (3.7 kB)
Collecting Jinja2>=3.1.2 (from Flask->-r requirements.txt (line 5))
  Using cached jinja2-3.1.4-py3-none-any.whl.metadata (2.6 kB)
Collecting itsdangerous>=2.2 (from Flask->-r requirements.txt (line 5))
  Using cached itsdangerous-2.2.0-py3-none-any.whl.metadata (1.9 kB)
Collecting click>=8.1.3 (from Flask->-r requirements.txt (line 5))
  Using cached click-8.1.7-py3-none-any.whl.metadata (3.0 kB)
Collecting blinker>=1.9 (from Flask->-r requirements.txt (line 5))
  Using cached blinker-1.9.0-py3-none-any.whl.metadata (1.6 kB)
Collecting packaging (from gunicorn->-r requirements.txt (line 8))
  Using cached packaging-24.2-py3-none-any.whl.metadata (3.2 kB)
Collecting charset-normalizer<4,>=2 (from requests->-r requirements.txt (line 9))
  Using cached charset_normalizer-3.4.0-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (34 kB)
Collecting idna<4,>=2.5 (from requests->-r requirements.txt (line 9))
  Using cached idna-3.10-py3-none-any.whl.metadata (10 kB)
Collecting urllib3<3,>=1.21.1 (from requests->-r requirements.txt (line 9))
  Using cached urllib3-2.2.3-py3-none-any.whl.metadata (6.5 kB)
Collecting certifi>=2017.4.17 (from requests->-r requirements.txt (line 9))
  Using cached certifi-2024.8.30-py3-none-any.whl.metadata (2.2 kB)
Collecting MarkupSafe>=2.0 (from Jinja2>=3.1.2->Flask->-r requirements.txt (line 5))
  Using cached MarkupSafe-3.0.2-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (4.0 kB)
Using cached bcrypt-4.2.1-cp39-abi3-manylinux_2_28_x86_64.whl (278 kB)
Using cached psutil-6.1.0-cp36-abi3-manylinux_2_12_x86_64.manylinux2010_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (287 kB)
Using cached pyotp-2.9.0-py3-none-any.whl (13 kB)
Using cached flask-3.1.0-py3-none-any.whl (102 kB)
Using cached Flask_Cors-5.0.0-py2.py3-none-any.whl (14 kB)
Using cached icmplib-3.0.4-py3-none-any.whl (30 kB)
Using cached gunicorn-23.0.0-py3-none-any.whl (85 kB)
Using cached requests-2.32.3-py3-none-any.whl (64 kB)
Using cached blinker-1.9.0-py3-none-any.whl (8.5 kB)
Using cached certifi-2024.8.30-py3-none-any.whl (167 kB)
Using cached charset_normalizer-3.4.0-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (143 kB)
Using cached click-8.1.7-py3-none-any.whl (97 kB)
Using cached idna-3.10-py3-none-any.whl (70 kB)
Using cached itsdangerous-2.2.0-py3-none-any.whl (16 kB)
Using cached jinja2-3.1.4-py3-none-any.whl (133 kB)
Using cached urllib3-2.2.3-py3-none-any.whl (126 kB)
Using cached werkzeug-3.1.3-py3-none-any.whl (224 kB)
Using cached packaging-24.2-py3-none-any.whl (65 kB)
Using cached MarkupSafe-3.0.2-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (23 kB)
Installing collected packages: urllib3, pyotp, psutil, packaging, MarkupSafe, itsdangerous, idna, icmplib, click, charset-normalizer, certifi, blinker, bcrypt, Werkzeug, requests, Jinja2, gunicorn, Flask, flask-cors
Successfully installed Flask-3.1.0 Jinja2-3.1.4 MarkupSafe-3.0.2 Werkzeug-3.1.3 bcrypt-4.2.1 blinker-1.9.0 certifi-2024.8.30 charset-normalizer-3.4.0 click-8.1.7 flask-cors-5.0.0 gunicorn-23.0.0 icmplib-3.0.4 idna-3.10 itsdangerous-2.2.0 packaging-24.2 psutil-6.1.0 pyotp-2.9.0 requests-2.32.3 urllib3-2.2.3


Ciao Gerd

**Comment by goerdi** [comment]
Hi !

Now im really confused.... i tried after your request and it seems to work.....
